ships/0005-local-source-upload.md

```diff
@@ -0,0 +1,262 @@
+---
+title: local-source-upload
+authors:
+  - @otaviof
+reviewers:
+  - @alicerum
+  - @adambkaplan
+approvers:
+creation-date: 2021-06-23
+last-updated: 2021-06-23
+---
+
+# Source Upload
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Summary
+
+Proposal to improve developer-experience by allowing to upload a local repository clone to the
+Kubernetes cluster, and use Shipwright Build Controller to produce a container-image out of it.
+
+As a developer, I can try Shipwright before opening a pull-request, and take advantage of the cluster
+resources to create the image. Effectively moving shipwright closer to the developer's inner-loop, in
+other words closer to the frequent activities while writing code ([1][InnerLoopA], [2][InnerLoopB]),
+in other words we are "moving left before _shipping-right_".
+
+## Motivation
+
+Shipwright Build-Controller allows teams to build container-images out of Git repositories, register
+and track their application builds in a Kubernetes native environment. This enhancement proposal
+extends this ability for developers to upload uncommitted changes and use the same machinery during
+the inner-loop as well.
+
+As a developer you can try out changes before opening a pull-request, make sure the final
+container-image will work out as expected, and effectively improve the developer experience.
+
+The approach proposed here is complementary to existing [EP][LocalSourceCodeEP] detailing local
+source upload to a container-registry and later on self-extracting the image before the actual
+container build process starts.
```

Comment:
Ok after re-reading your updates and your EP refs, there are in fact connections to two existing EP's.  `LocalSourceCodeEP` and `RemoteArtifactsEP`.  That latter in fact ties into your API example.  I *think* the former pertains to the notion of being able to swap different "source upload adapters" (i.e. kubectl cp vs. the registry approach).

Assuming I'm close here, I feel like a statement or two giving a high level description somewhere along these lines (or different details if you feel my summaries and somewhat off), explaining the relationship of your new EP here, with these other 2, is a good way to kick things off.

Then below the more detailed refs to these 2 EPs that you already have build upon the "introduction" I'm asking for here.

Reply:
ok @otaviof pointed me to where below when he discusses the CLI element of the flow, the user will have a choice on which upload option to use.

I'm fine with capturing the "swapability" of all this there and just citing the "complementary" element with this EP to the other EP here at the beginning.